main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"golang.org/x/crypto/sha3"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	EncryptionProviderConfigShake256SecretName      = "encryption-provider-config-shake256"
+	EncryptionProviderConfigShake256SecretNamespace = "kube-system"
+
+	EnvNodeName = "NODE_NAME"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	_ = clientgoscheme.AddToScheme(scheme)
+
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	ctx := context.TODO()
+	// flags
+	var encryptionConfigFilePath string
+	flag.StringVar(&encryptionConfigFilePath, "encryption-config-file-path", "/etc/kubernetes/encryption/k8s-encryption-config.yaml", "The path to the encryption config file.")
+
+	// read the encryption config file
+	f, err := ioutil.ReadFile(encryptionConfigFilePath)
+	if err != nil {
+		fmt.Printf("ERROR: failed to read file %s\n", encryptionConfigFilePath)
+		panic(err)
```

Comment:
nit, what about using os exit instead of panic?

Reply:
what would be the advantage of it?

Reply:
no huge advantage here but I think it might be easier to check/test if you would add unit tests.